### ISSUE TITLE
Heading block controls: show buttons to set heading size

### ISSUE THREAD

**Issue by karmatosed** [WordPress/gutenberg#74]
From prototype testing:

> “title” is an H (heading), but tagged as an H2 when I inspect the page source. What if I want an H1 or H3 etc?

https://make.wordpress.org/design/2017/02/09/initial-gutenberg-prototype-editor-testing/#comment-25060

**Comment by jasmussen** [comment]
This is <a href="https://github.com/WordPress/gutenberg#mockups">addressed in the mockups</a>. Relabelling the ticket to serve as a design task, hope that's okay ❤️

**Comment by jasmussen** [comment]
The current mockups provide block level controls to change the type/size of heading, with "Heading" still being a single block:

<img src="https://raw.githubusercontent.com/WordPress/gutenberg/master/mockups/Headings.png" />

Every level of heading, 1-6, are shown here so you can in only two clicks set the size of a heading (click block, click size). Alternate ideas explored included up/down arrows to bump or reduce the size of a heading, but these would require multiple clicks. 

But the H1/H2/H3 ... buttons aren't very inviting, and feel very "techy". Is there a better way to solve this? A better icon design? Can we show 6 levels of sizes in six different icons? Should we only show sizes 1-3 (with sized icons) and show the rest in a dropdown? 

Input welcome!

**Comment by shaunandrews** [comment]
Here's a quick idea for showing fewer options, but with an arrow to expand to more:

![heading selector](https://cloud.githubusercontent.com/assets/191598/23462802/55240ab0-fe5d-11e6-82a2-4472016bad5b.gif)

And this video shows an idea to show a preview of the heading size (or whatever ever style) rather than a generic H1 icon:

![heading selector with preview](https://cloud.githubusercontent.com/assets/191598/23462900/a238c1e2-fe5d-11e6-9237-d2cada55430f.gif)

**Comment by jasmussen** [comment]
So in the spirit of exploration, here's a crazy idea:

![headings alternate type switcher](https://cloud.githubusercontent.com/assets/1204802/23608816/6db4f41a-026b-11e7-9bc0-fad5815407e0.png)

The icons are a little fuzzy and unpolished, but this embraces the fact that H2 and H3 are technically separate blocks (as opposed to how we've seen it so far as a Heading block with the size being the defining difference). 

Basically you pick the size of you heading here. Or you pick text, or quote or whatever it needs to be. The formatting toolbar is then contextual to the type.

**Comment by jasmussen** [comment]
Here's a more basic one, which is more akin to Shauns version above:

![headings](https://cloud.githubusercontent.com/assets/1204802/23610118/a52c23ea-0271-11e7-816e-fa771e090610.png)

**Comment by jasmussen** [comment]
Closing in favor of #313
